core/modules/block/block.module

```diff
@@ -89,7 +89,7 @@ function block_block_info() {
   foreach ($config_names as $config_name) {
     $block = config_get($config_name);
     $blocks[$block['delta']] = $block;
-    $blocks[$block['delta']]['description'] = t('A reusable custom block.');
+    $blocks[$block['delta']]['description'] = isset($block['description']) && strlen($block['description']) != 0 ? $block['description'] : t('A reusable custom block.');
```

Comment:
crap, missed this one when hitting "Approve" button :/

Replace `isset($block['description']) && strlen($block['description']) != 0` with a cleaner `!empty($block['description'])`